frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx

```diff
@@ -1,131 +1,56 @@
 import * as React from 'react';
-import * as _ from 'lodash';
 import { useTranslation } from 'react-i18next';
-
-import * as classNames from 'classnames';
-import {
-  InputGroup,
-  FormGroup,
-  TextInput,
-  FormSelect,
-  FormSelectOption,
-  Button,
-  ValidatedOptions,
-  Tooltip,
-} from '@patternfly/react-core';
-import { global_palette_blue_300 as blueInfoColor } from '@patternfly/react-tokens/dist/js/global_palette_blue_300';
-import { PencilAltIcon, EyeIcon, EyeSlashIcon } from '@patternfly/react-icons';
-
-import { setEncryptionDispatch, parseURL, kmsConfigValidation, EncryptionDispatch } from './utils';
-import { advancedKMSModal } from '../modals/advanced-kms-modal/advanced-kms-modal';
-import { InternalClusterState, ActionType } from '../ocs-install/internal-mode/reducer';
-import { KMSProviders } from '../../constants';
-import { KMSConfig } from '../../types';
-import { State } from '../ocs-install/attached-devices-mode/reducer';
-import { WizardState } from '../create-storage-system/reducer';
-
+import { FormGroup, FormSelect, FormSelectOption } from '@patternfly/react-core';
+import { useFlag } from '@console/shared/src/hooks/flag';
+import { ValutConfigure } from './vault-config';
+import { IbmKmsConfigure } from './ibm-kms-config';
+import { EncryptionDispatch, KMSConfigureProps } from './providers';
+import { isLengthUnity } from './utils';
+import { GUARDED_FEATURES } from '../../features';
+import { ProviderNames } from '../../types';
 import './kms-config.scss';
 
+const KMSProviders = [
+  {
+    name: 'Vault',
```

Comment:
Name is used as a label in https://github.com/openshift/console/pull/10475/files#diff-ae3d56459b2491d83b6fe9502d3c74b3910fc605496fef5d596dcf713a097af6R72, need to localize. one way is to return KMSProviders from the function which takes Tfunction arg as a parameter.

Reply:
But, this name is the name of the product. Just like we have `OpenShift Container Storage`. Not sure if we want to localize product's name ??